client/verta/verta/pipeline/_pipeline_step.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+
+from verta._internal_utils._utils import Configuration, Connection
+from verta.registry.entities import RegisteredModel, RegisteredModelVersion
+
+
+class PipelineStep:
+    """Object representing a single step to be run within an inference pipeline.
+
+    Parameters
+    ----------
+    name : str
+        Name of the step, for use within the scope of the pipeline only.
+    registered_model_version : :class:`~verta.registry.entities.RegisteredModelVersion`
+        Registered model version to run for this step.
+    predecessors : list, set, tuple, optional
+        List, set, or tuple of unique PipelineStep objects whose outputs will be treated as
+        inputs to this step. If not included, the step is assumed to be an initial step.
+
+    Attributes
+    ----------
+    name : str
+        Name of the step within the scope of the pipeline.
+    registered_model_version : :class:`~verta.registry.entities.RegisteredModelVersion`
+        Registered model version run by this step.
+    predecessors : set
+        Set of PipelineSteps whose outputs will be treated as inputs to this step.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        registered_model_version: RegisteredModelVersion,
+        predecessors: Optional[
+            Union[List["PipelineStep"], Set["PipelineStep"], Tuple["PipelineStep"]]
+        ] = None,  # Optional because it could be the first step with no predecessors
+    ):
+        self._name = self.set_name(name)
+        self._registered_model_version = self.set_registered_model_version(
+            registered_model_version
+        )
+        self._predecessors = self.set_predecessors(predecessors)
+        self._registered_model: RegisteredModel = self._get_registered_model()
```

Comment:
```python
self._registered_model_version = self.set_registered_model_version(
    registered_model_version
)
```

already sets `self._registered_model`. So we're fetching the RM twice during `__init__()` (which also results in two `got existing RegisteredModel` print statements) which may be a little janky. But we can address this later.

Reply:
Fixing this won't be a huge deal, but may be best left to another [ticket](https://vertaai.atlassian.net/browse/VRD-1102)